modules/infra/control/control_output.c

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr_control.h>
+#include <gr_control_output.h>
+#include <gr_graph.h>
+#include <gr_log.h>
+#include <gr_macro.h>
+#include <gr_mbuf.h>
+
+#include <event2/event.h>
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+
+#include <pthread.h>
+#include <stdatomic.h>
+#include <stdlib.h>
+
+static struct rte_ring *ctrlout_ring;
+static struct event *ctrlout_ev;
+
+static bool tshutdown;
+static pthread_t thread_id;
+static pthread_cond_t cv;
+static pthread_mutex_t mp;
+
+int control_output_push(struct gr_control_output_msg *msg) {
+	return rte_ring_enqueue_elem(ctrlout_ring, msg, sizeof(*msg));
+}
+
+static void poll_control_output_ring(evutil_socket_t, short, void *) {
+	struct gr_control_output_msg msg[RTE_GRAPH_BURST_SIZE];
+	int n;
+
+	while ((n = rte_ring_dequeue_burst_elem(
+			ctrlout_ring, msg, sizeof(msg[0]), ARRAY_DIM(msg), NULL
+		))
+	       > 0) {
```

Comment:
Automatic formatting is a bit ugly here...

I wish we could do better but I don't see how.

Reply:
I'll remove the dequeue burst, and will dequeue 1 el at a time.
This shouldn't be a critical part of the code considering the number of elements.